File: api-delpi/app/domain/entities/audit_5s/audit_5s.py
from dataclasses import dataclass, asdict
from typing import Optional


@dataclass
class Audit5S:
    id: str
    date: Optional[str]
    average_line_score: Optional[float]
    evaluated_area: Optional[str]
    auditor: Optional[str]
    audited: Optional[str]
    inspection_number: Optional[str]
    shift: Optional[str]
    branch: Optional[str] = None

    def to_dict(self) -> dict:
        return asdict(self)

File: plugins/dashboard-quality/src/types/audit5s.ts
export type Audit5sSummaryParams = {
  start_date?: string;
  end_date?: string;
  branch?: string;
};

export type Audit5s = {
  id: string;
  date: string | null;
  average_line_score: number | null;
  evaluated_area: string | null;
  auditor: string | null;
  audited: string | null;
  inspection_number: string | null;
  shift: string | null;
  branch: string | null;
};

export type Audit5sSummary = {
  start_date: string | null;
  end_date: string | null;
  average_score: number;
  list_audits: Audit5s[];
};
